equinox-projector/Consumer/Examples.fs

```diff
@@ -0,0 +1,145 @@
+    member __.TryDecode(streamName, spanJson) = seq {
+        let tryExtractCategory (stream : string) = stream.Split([|'-'|], 2, StringSplitOptions.RemoveEmptyEntries)
+        let events = JsonConvert.DeserializeObject<RenderedSpan>(spanJson) |> RenderedSpan.enumEvents
+        match tryExtractCategory streamName with
+        | [| "Favorites"; _ |] -> yield! events |> Seq.choose (tryDecode log streamName Favorites.codec >> Option.map Faves)
+        | [| "SavedForLater"; _ |] -> yield! events |> Seq.choose (tryDecode log streamName SavedForLater.codec >> Option.map Saves)
+        | [| category; _ |] -> yield Category (category, Seq.length events)
+        | _ -> yield Unclassified streamName }
+
+type Processor() =
+    let mutable favorited, unfavorited, saved, cleared = 0, 0, 0, 0 
+    let cats, keys = CatStats(), ConcurrentDictionary()
+
+    member __.DumpStats(log : ILogger) =
+        log.Information("Favorited {f} Unfavorited {u} Saved {s} Cleared {c}", favorited, unfavorited, saved, cleared)
+        favorited <- 0; unfavorited <- 0; saved <- 0; cleared <- 0; cats.Clear(); keys.Clear()
+    member __.Handle = function
+        | Faves (Favorites.Favorited _) -> Interlocked.Increment &favorited |> ignore
+        | Faves (Favorites.Unfavorited _) -> Interlocked.Increment &unfavorited |> ignore
+        | Saves (SavedForLater.Added e) -> Interlocked.Add(&saved,e.skus.Length) |> ignore
+        | Saves (SavedForLater.Removed e) -> Interlocked.Add(&cleared,e.skus.Length) |> ignore
+        | Saves (SavedForLater.Merged e) -> Interlocked.Add(&saved,e.items.Length) |> ignore
+        | Category (cat,count) -> lock cats <| fun () -> cats.Ingest(cat, int64 count)
+        | Unclassified messageKey -> keys.TryAdd(messageKey, ()) |> ignore
+
+open Confluent.Kafka
+open Jet.ConfluentKafka.FSharp
+
+type BatchingSync =
+    /// Starts a consumer that consumes a topic in a batched mode, based on a source defined by `cfg`
+    /// Processing runs as a single Async computation per batch, which can work well where parallism is not relevant
+    static member Start(cfg: KafkaConsumerConfig) =
+        let log = Log.ForContext<BatchingSync>()
+        let interpreter = MessageInterpreter()
+        let handleBatch (msgs : ConsumeResult<_,_>[]) = async {
+            let processor = Processor()
+            for m in msgs do
+                for x in interpreter.TryDecode(m.Key, m.Value) do
+                    processor.Handle x
+            processor.DumpStats log }
+        BatchedConsumer.Start(log, cfg, handleBatch)
+        
+type Streaming =
+    /// Starts a consumer that consumes a topic in streamed mode
+    /// StreamingConsumer manages the parallelism, spreading individual messages out to Async tasks
+    /// Optimal where each Message naturally lends itself to independent processing with no ordering constraints
+    static member Start(cfg: KafkaConsumerConfig, degreeOfParallelism: int) =
```

Comment:
@szer this is the usage pattern for https://github.com/jet/Jet.ConfluentKafka.FSharp/pull/33